### PR TITLE
Create Basic Data Visualizations with Python-Workshop

### PR DESCRIPTION
**Pull request by maddieWillett-boop**
name: 'Basic Data Visualizations with Python'
description: 'Create basic graphs and data visualizations using Python!'
author: '@maddieWillett-boop'
[Link to Demo!](https://hackclub.slack.com/archives/C01A9V5LPD0/p1601172667002400)

### CODE REVIEW

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:9]: could you host this on Hack Club's CDN?

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:12]:
```suggestion
```

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:32]:
```suggestion

```

I dont think we need these

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:313]: could we delete

**Comment by sampoder** [comment]
hey @maddieWillett-boop could you make the Markdown a bit cleaner with smth like: https://www.markdownbeautifier.com/#

File: workshops/python and data visualizations/README.md
Comment (sampoder) [reply workshops/python and data visualizations/README.md:9]: oh also add alt text

File: workshops/python and data visualizations/README.md
Comment (maddieWillett-boop) [reply workshops/python and data visualizations/README.md:311]: which part?

Comment (sampoder) [reply workshops/python and data visualizations/README.md:313]: the extra new lines

Comment (maddieWillett-boop) [reply workshops/python and data visualizations/README.md:284]: @samepoder done

Comment (sampoder) [reply workshops/python and data visualizations/README.md:284]: cool! thanks, same (jk)

**Comment by sampoder** [comment]
coz my notes above are simply formatting im going to do a content review rn

**Review by sampoder — COMMENTED** [review]
Did a content review, uploading soon @ https://youtu.be/6deLs_fFZ2s

Let me know if you've got any questions :D

**Comment by maddieWillett-boop** [comment]
@sampoder Made a bunch of edits! Hopefully made it clearer, added some code descriptions, etc. Let me know what you think! :)

Also... [Chicken Little](https://en.wikipedia.org/wiki/Chicken_Little_(2005_film)) is a movie haha you made me laugh so hard w/ ur narrations of things.

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:1]: lets move this below the header

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:2]:
```suggestion
# Create basic graphs and data visualizations with python!
```

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:4]: what about a stonks meme to mix it up?

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:5]:
```suggestion

Follow along with this workshop to gain some data science skills! You will use python to create basic graphs and visualizations with real data. You can hand draw a graph anytime you want, but why not learn how to code one! You will learn about matplotlib, pandas, and numpy, so you can create any graph or vizualizaiton you dream of!
```

File: workshops/python and data visualizations/README.md
Comment (sampoder) [inline workshops/python and data visualizations/README.md:6]:
```suggestion

```

**Comment by sampoder** [comment]
in no way is this complete but: https://github.com/maddieWillett-boop/hackclub/pull/1 has some formatting changes that could be added

**Comment by maddieWillett-boop** [comment]
@sampoder made some more formatting changes, do you spot anything else that needs to be fixed? When you say "in no way is this complete" what should I work on to make it closer to completion?

**Comment by sampoder** [comment]
Wait whend I say that

**Comment by sampoder** [comment]
I doubt I meant it in that way.... lol

**Comment by maddieWillett-boop** [comment]
@sampoder Oh I did not mean that in a bad way haha :) , I just was wondering if there is anything else?

**Comment by sampoder** [comment]
main thing rn is we need the demos

**Comment by maddieWillett-boop** [comment]
From the club meeting? I uploaded a video to the workshop-skunk works channel. But let me try again...

**Comment by maddieWillett-boop** [comment]
can you see it now?

**Comment by sampoder** [comment]
I mean the 3 examples for building on top of the workshop

**Comment by maddieWillett-boop** [comment]
ok if you scroll down to the very bottom of the workshop there is a "add on" and "learn more" section that have 3+ links and ideas in each part.

**Comment by sampoder** [comment]
Yep, so with those I think we want them to show progression from this specific workshop, correct me if I'm wrong they seem to be unrelated

**Comment by maddieWillett-boop** [comment]
Ok, so should I show some examples that I made with the add on resources? 

I would say they are related to the workshop because with those resources all you have to do is add a couple lines to the code that we made above. One of the resources teaches you how to import csv files and it will plot points from the real data into graph instead of making up plotting points. Also, looking up to use google trends and incorporate that into a graph,  another source that has code and teaches you how to create different graphs (how to add colors, shapes, dot plots, pie charts, etc.) And a link to another “workshop” that uses the same IDE and code we learned to make data visualizations with Covid-19 Data. Then in the learn more section there are links that teach you more about the libraries we used and all the things you can create amongst them. For instance, matplotlib, pandas, and numpy. So I think that they all relate back to the workshop, and encourage them to add more/personalize the basic graphs we made in the workshop.

**Comment by maddieWillett-boop** [comment]
I think it shows progression because you can really build upon the simple graphs and plots we made with the resources, it brings it up to another level. What the goal of the workshop was to teach the basics and the add ons use the basics you learn as a foundation and add more to make it more complicated and intersting.  For instance, in my club during the time were they were adding on to what they made one member imported google trends into his project and made a graph that used real google trend data to see the correlation of tik tok and popularity over time, another one personalized it to show comparisons from his life in the United States and Inda, and another learned how to import a csv file from the resources and made a data visualization that showed the population of the United States over time.

**Comment by maddieWillett-boop** [comment]
But maybe I am misinterpreting it, let me know what you think...

**Comment by sampoder** [comment]
k, I don't think I phrased it correctly. The demos (based on all the other workshops) need to be directly building on top of the project, view example: https://workshops.hackclub.com/sound_galaxy/#hacking

**Comment by sampoder** [comment]
now I've asked Zach to clarify if im getting smth wrong (just started helping out with workshop review)

**Comment by sampoder** [comment]
ohhh also when I said in no way is this complete I was referring to **my** pr, not this pr.... could you merge that pr?

**Comment by sampoder** [comment]
https://github.com/maddieWillett-boop/hackclub/pull/1

**Comment by maddieWillett-boop** [comment]
Ohhh haha I see now okay I will make those edits @sampoder, also I still think that the demos are directly building on top of the project lol maybe my edit is that it needs to be that I show the exact code as well (kind of like how I did w/ examples in the workshop, and how it is in the sound galaxy workshop)?

**Comment by sampoder** [comment]
yep, how it is in the sound galaxy one, thank you!

**Comment by maddieWillett-boop** [comment]
@sampoder fixed it, now there are examples :)

**Comment by maddieWillett-boop** [comment]
Anything else that should be edited?

**Comment by sampoder** [comment]
will review in a bit

**Comment by maddieWillett-boop** [comment]
ok sounds good

**Comment by maddieWillett-boop** [comment]
@sampoder do you think it is ready for the next steps?

**Comment by sampoder** [comment]
I kinda want a second person's perspective, @faisalsayed10 would you mind taking a look?

**Comment by faisalsayed10** [comment]
Yep, taking a look into it rn.

File: workshops/python and data visualizations/README.md
Comment (faisalsayed10) [inline workshops/python and data visualizations/README.md:9]:
```suggestion
Get started by going to [Jupyter](https://jupyter.org/try). When you get to the website you will see something like this...
```

Comment (faisalsayed10) [reply workshops/python and data visualizations/README.md:9]: "Get started by going to Jupyter" looks more good in my opinion

File: workshops/python and data visualizations/README.md
Comment (faisalsayed10) [inline workshops/python and data visualizations/README.md:154]: a full stop or an exclamation at the end here?

File: workshops/python and data visualizations/README.md
Comment (faisalsayed10) [inline workshops/python and data visualizations/README.md:158]:
```suggestion
*Make sure libraries are uploaded, look back @ Step Two if needed*.
```

**Review by faisalsayed10 — COMMENTED** [review]
These are the very few changes and overall this workshop is great! :+1:

**Comment by maddieWillett-boop** [comment]
Awesome, thank you @faisalsayed10!

**Comment by maddieWillett-boop** [comment]
Made those updates, what should I do next?

**Comment by faisalsayed10** [comment]
@sampoder we can ask for the final review now?

**Comment by maddieWillett-boop** [comment]
@sampoder @faisalsayed10 sounds good to me, just lmk what you want me to edit :)

**Comment by maddieWillett-boop** [comment]
@sampoder do you have an idea of what I should do next (editing wise, etc.) is it ready for the final review?

**Comment by sampoder** [comment]
yep, waiting for final review (cc: @zachlatta)

**Comment by zachlatta** [comment]
Yes, thanks. This is next in line for my review.

On Oct 6, 2020, at 22:03, Sam Poder <notifications@github.com> wrote:

﻿

yep, waiting for final review (cc: @zachlatta <https://github.com/zachlatta>
)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/hackclub/hackclub/pull/1332#issuecomment-704648122>, or
unsubscribe

File: workshops/python and data visualizations/README.md
Comment (bezlin6mechminerz) [inline workshops/python and data visualizations/README.md:5]:
```suggestion
*Follow along with this workshop to gain some data science skills! You will use python to create basic graphs and visualizations with real data. You can hand draw a graph anytime you want, but why not learn how to code one! You will learn about matplotlib, pandas, and numpy, so you can create any graph or visualization you dream of!*
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:1]: Please re-host on the #cdn channel :)

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:2]: Please rename the folder to `python_dataviz` or something—remove the spaces though

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:3]: Please add the metadata as requested in the bounty guidelines to the start of the file

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:5]: The start of your workshop should lay out a specific thing you’re going to make, with a link &/or screenshot of the live demo, as well as the source code. Even if folks deviate from the workshop (which we hope they will) to take it in their own direction, they should be able to follow it with no changes & get exactly what you did.

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:40]:
```suggestion
```python
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:65]: Add alt text

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:87]: alt text

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:121]:
```suggestion
### Okay, now let’s get back to it...
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:165]:
```suggestion
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:170]:
```suggestion
Here’s a hint:
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:176]:
```suggestion
*Still stuck? Where the underscores are, replace with `fig,ax= plt.subplots()`.*
```

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:210]: alt text here & below

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:235]: You should have screenshots throughout of what your graph should look like if you’ve followed the code correctly

File: workshops/python and data visualizations/README.md
Comment (lachlanjc) [inline workshops/python and data visualizations/README.md:283]:
```suggestion
```

**Review by lachlanjc — CHANGES_REQUESTED** [review]
This is an awesome start!!! I’ve made a bunch of specific suggestions & requests here, the ones at the beginning which we’ll definitely need in order to to merge. I think my primary piece of feedback is that you should focus it at the beginning a little more on a specific end product: we’re going to do exactly x using Python & Jupyter, with a screenshot of what you’ll have made by the end. We have a lot of fun GIFs & graphics throughout, but I’d replace or add to a few of those with screenshots of the project at each stage, so it’s clear if folks are on track or not. I think the last section also needs more fleshing out/context, as in: here’s how to share your resulting project, then here’s some suggestions for where to take your project, not just the links to documentation/resources (which are awesome). This is 2/3 of the way there, & I think with those beginning edits & a bit of clarification at the beginning & end we’ll push it to completion. Super appreciate your submission!

EDIT: Going to close this for now since we’re clearing out the PRs but if you’re interested, once you’ve focused the article on a specific project, go ahead & make a new PR & we’re happy to review. Thanks!

**Comment by lachlanjc** [comment]
(Pulling this comment down so it’s easier to find.) Closing this for now since we’re clearing out the PRs here, but if you’re interested, once you’ve focused the article on a specific project, go ahead & make a new PR & we’d be super happy to review. Thanks a bunch!

**Comment by maddieWillett-boop** [comment]
@lachlanjc submitted new pull request :)

**Pull request closed without merging**
